packages/types/src/pnum.ts

```diff
@@ -0,0 +1,115 @@
+import { BigNumber } from 'bignumber.js';
+import { round } from '@penumbra-zone/types/round';
+import { LoHi, joinLoHi, splitLoHi } from '@penumbra-zone/types/lo-hi';
+import { Amount } from '@penumbra-zone/protobuf/penumbra/core/num/v1/num_pb';
+import { ValueView } from '@penumbra-zone/protobuf/penumbra/core/asset/v1/asset_pb';
+import { getAmount, getDisplayDenomExponentFromValueView } from '@penumbra-zone/getters/value-view';
+import { removeTrailingZeros } from '@penumbra-zone/types/shortify';
+
+/**
+ * pnum (penumbra number)
+ *
+ * In Penumbra a number can be in the form of a base unit (bigint, LoHi, Amount, ValueView)
+ * or a number with decimals for display purposes (string, number)
+ *
+ * This function handles all these cases automatically internally
+ * - when input is a bigint, LoHi, Amount, or ValueView, it is assumed to be in base units
+ * - when input is a string or number, it is multiplied by 10^exponent and converted to base units
+ *
+ * Likewise for all methods, the outputs are
+ * - in base units for bigint, LoHi, Amount and ValueView
+ * - in display form with decimals for string and number
+ *
+ * @param input
+ * @param exponent
+ */
+function pnum(
+  input: string | number | LoHi | bigint | Amount | ValueView | undefined,
```

Comment:
question: what are the usecases for an `undefined` input?

Reply:
This: 
<img width="966" alt="image" src="https://github.com/user-attachments/assets/d0df1de2-2c16-411b-891a-688ee0369ab4">

I guess you can make an argument against it, what do you think?